apps/review-host/check.sh

```
@@ -0,0 +1,155 @@
+
+# The store listing promises "review the diff from your phone", so an empty diff
+# is a listing mismatch, not a cosmetic gap. It has been empty twice: once when
+# the worktrees were rebuilt as bare branches, and once when the demo repos lost
+# refs/remotes/origin/HEAD and the base could not resolve at all.
+if [ -n "$WS_JSON" ]; then
+  WS_IDS=$(printf '%s' "$WS_JSON" | python3 -c 'import json,sys
+d = json.load(sys.stdin)["result"]["data"]["json"]
+print(" ".join(w["id"] for w in d if w.get("type") != "main"))' 2>/dev/null)
+  if [ -n "$WS_IDS" ]; then
+    NODIFF=$(ssh_box "sudo sh -c 'for W in $WS_IDS; do n=\$(curl -s -m 15 -X POST -H \"Authorization: Bearer $SECRET\" -H \"content-type: application/json\" --data \"{\\\"json\\\":{\\\"workspaceId\\\":\\\"\$W\\\"}}\" http://127.0.0.1:48800/trpc/git.listCommits | grep -c hash); [ \"\$n\" = \"0\" ] && echo \$W; done'")
+    [ -z "$NODIFF" ] && note "diffs: every demo workspace has one" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)